component/common/loki/types.go

```diff
@@ -27,6 +29,14 @@ const finalEntryTimeout = 5 * time.Second
 // communication.
 type LogsReceiver chan Entry
 
+// TODO(piotr): Since LogsReceiver is not an interface, we can't extend it...
+func (l LogsReceiver) RiverTokenize() []builder.Token {
+	return []builder.Token{{
+		Tok: token.STRING,
+		Lit: "loki.write.FIX_ME.receiver",
+	}}
+}
+
```

Comment:
@erikbaranowski I ran into this issue when trying to follow the pattern you've used for prometheus config conversion... I cannot extend `LogsReceiver` with `RiverTokenize` because it's not an interface.

I am happy to implement a refactor in a separate PR for changing `type LogReceiver chan Entry` into an interface, something like:
```
type LogReceiver interface {
    Chan() chan Entry
}
```

what do you think?

Reply:
we spoke offline about trying it as an interface since other tricks don't line up exactly in this case.